plugins/dig-dug.cpp

```diff
@@ -0,0 +1,223 @@
+/*
+ * Simulates completion of dig designations.
+ */
+
+#include "DataFuncs.h"
+#include "PluginManager.h"
+#include "TileTypes.h"
+
+#include "modules/Maps.h"
+
+#include <df/tile_designation.h>
+#include <df/tile_occupancy.h>
+#include <df/world.h>
+#include <df/map_block.h>
+
+DFHACK_PLUGIN("dig-dug");
+REQUIRE_GLOBAL(world);
+
+using namespace DFHack;
+
+// returns true iff tile is in map bounds and was hidden before this function
+// unhid it.
+static bool unhide(int32_t x, int32_t y, int32_t z) {
+    // ensures coords are in map bounds and ensures that the map block exists
+    // so we can unhide the tiles
+    if (!Maps::ensureTileBlock(x, y, z))
+        return false;
+
+    df::tile_designation &td = *Maps::getTileDesignation(x, y, z);
+    if (!td.bits.hidden)
+        return false;
+
+    td.bits.hidden = false;
+    return true;
+}
+
+// unhide self and adjacent tiles if hidden and flood fill unhidden state
+static void flood_unhide(int32_t x, int32_t y, int32_t z) {
```

Comment:
Ah, I hadn't seen that. `reveal`'s logic does look more sophisticated, and I am always happy to *not* have to write code : )

How to share, though? Going through the Lua layer on every tile risks being slow when the designated area is large. A large designated area is unlikely, though, and, unlike the otherwise similar situation for `blueprint`, I can't think of a use case where someone could possibly be running a large designated area so frequently that it would matter.

I could also keep track of adjacent dug tiles and just run once per group.

Reply:
done. I ended up doing all the unrevealing at the end so we wouldn't get dueling MapCache instances.